conans/model/settings.py

```diff
@@ -172,7 +172,8 @@ def __init__(self, definition=None, name="settings", parent_value="settings"):
                       for k, v in definition.items()}
 
     def serialize(self):
-        return {k: v.value for k, v in self._data.items()}
+        # Returning all the information available, including the sub-settings
+        return self.items()
```

Comment:
But this change is not returning a "string" object, but a ``SettingsItem`` one.
The idea is that ``serialize()`` methods return hierarchy of python primitives, strings, dicts, lists, numbers...